lib/privacy_pass_redeemer.rb

```diff
@@ -0,0 +1,111 @@
+class PrivacyPassRedeemer
+  HEADER_FORMAT = /^PrivateToken token=(?<token>.*)$/
+  # See https://www.ietf.org/archive/id/draft-ietf-privacypass-protocol-04.html#name-token-type
+  # See https://www.ietf.org/archive/id/draft-ietf-privacypass-protocol-04.html#section-6.3
+  # struct {
+  #   uint16_t token_type = 0x0002
+  #   uint8_t nonce[32];
+  #   uint8_t challenge_digest[32];
+  #   uint8_t token_key_id[32];
+  #   uint8_t authenticator[Nk];
+  # } Token;
+  TOKEN_TYPE_LENGTH = 1
```

Comment:
should this be 2 since it's a `uint16_t` as opposed to a uint8?

Reply:
No, because we're checking the length unpacked data and we're unpacking this via the uint16 directive (`n`)